src/charm.py

```diff
@@ -58,6 +57,15 @@ def _on_config_changed(self, event):
         Prometheus config file is regenerated, pushed to the workload
         container and Prometheus is restarted.
         """
+        if isinstance(event, PebbleReadyEvent):
```

Comment:
This is unusual. I do not see why it makes sense in a "Config Changed" event handler to check if the event is a "Pebble Ready" event.

Reply:
It makes sense because you're observing both event types with the same handler. It would probably be better to make two separate handlers and move the shared implementation out into the implementation class, though.